`softplus`: port to structured

Freey0 opened the pull request:
Stack from [ghstack](https://github.com/ezyang/ghstack):
* #58050 Port silu to structured
* #57623 Port softshrink to structured
* #57622 Port hardsigmoid to structured
* #57621 Port leaky_relu to structured
* **#57620 Port softplus to structured**
* #57619 Port elu to structured

Differential Revision: [D28224705](https://our.internmc.facebook.com/intern/diff/D28224705)

Freey0 committed:
Port softplus to structured

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28224705).

ezyang commented:
```
May 05 16:28:13 ======================================================================
May 05 16:28:13 FAIL [0.007s]: test_softplus_inplace_overlap_meta (__main__.TestNNDeviceTypeMETA)
May 05 16:28:13 ----------------------------------------------------------------------
May 05 16:28:13 Traceback (most recent call last):
May 05 16:28:13   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_utils.py", line 1030, in wrapper
May 05 16:28:13     method(*args, **kwargs)
May 05 16:28:13   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 292, in instantiated_test
May 05 16:28:13     result = test_fn(self, *args)
May 05 16:28:13   File "/opt/conda/lib/python3.6/site-packages/torch/testing/_internal/common_device_type.py", line 710, in efail_fn
May 05 16:28:13     slf.fail('expected test to fail, but it passed')
May 05 16:28:13 AssertionError: expected test to fail, but it passed
May 05 16:28:13 
```

Freey0 committed:
Update on "Port softplus to structured"


Differential Revision: [D28224705](https://our.internmc.facebook.com/intern/diff/D28224705)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28224705).

Freey0 committed:
Update on "Port softplus to structured"


Differential Revision: [D28224705](https://our.internmc.facebook.com/intern/diff/D28224705)

[ghstack-poisoned]

ezyang commented:
@ezyang has imported this pull request.  If you are a Facebook employee, you can view this diff [on Phabricator](https://www.internalfb.com/diff/D28224705).
